### ISSUE TITLE
KafkaAdmin client as part of KafkaAutoConfiguration

### ISSUE THREAD

**Issue by Jcamilorada** [spring-projects/spring-boot#11517]
[AdminClient](https://kafka.apache.org/10/javadoc/index.html?org/apache/kafka/clients/admin/AdminClient.html) provides useful admin operations over kafka. Consider adding as part of kafka auto-configuration beans. 

- One can argue that this functionality need to be wrapped by [KafkaAdmin](https://github.com/spring-projects/spring-kafka/blob/master/spring-kafka/src/main/java/org/springframework/kafka/core/KafkaAdmin.java)

- created as part discussion [11435](https://github.com/spring-projects/spring-boot/issues/11435) pull request discussion

**Comment by Jcamilorada** [comment]
created as part discussion [11435](https://github.com/spring-projects/spring-boot/issues/11435) pull request discussion

**Comment by snicoll** [comment]
paging @garyrussell to get more feedback on this.

**Comment by garyrussell** [comment]
A `KafkaAdmin` `@Bean` is [already autoconfigured](https://github.com/spring-projects/spring-boot/blob/master/spring-boot-project/spring-boot-autoconfigure/src/main/java/org/springframework/boot/autoconfigure/kafka/KafkaAutoConfiguration.java#L134).

**Comment by garyrussell** [comment]
If an application needs access to an `AdminClient` directly, e.g. 
to use features not exposed by the KafkaAdmin, it can use `AdminClient.create(kafkaAdmin.getConfig())`.

But I don't have a strong objection to adding the bean if it's felt necessary.

**Comment by snicoll** [comment]
Unless this is really necessary, I'd rather avoid exposing it. @Jcamilorada perhaps you could rework your original PR with that suggestion?

**Comment by garyrussell** [comment]
I looked at the health indicator PR.

One benefit of it being a `@Bean` is that it implements `AutoCloseable`, so Spring will close the connection, when the context is `stop()` ped, if an application forgets to do so.

With my suggestion above, the app is responsible for doing his own close.

On the other hand, I wonder if we really should keep open a long-lived admin connection for health checks; perhaps it would be better to use a new `AdminClient` each time (and close the resources).

**Comment by Jcamilorada** [comment]
@garyrussell @snicoll  thanks, I will update pull request to use new `AdminClient` each time, probably test will require `KafkaEmbedded` as well.

**Comment by snicoll** [comment]
Alright, thanks both!

**Comment by massivespace** [comment]
I'm trying to access the admin client's topic list for actuator health endpoint, but the suggestion above to use AdminClient.create(kafkaAdmin.getConfig()) does not work for me.

I'm using kotlin with spring boot, so it's a bit different: AdminClient.create(kafkaAdmin?.configurationProperties), but the list of configuration properties is only a single item, a default boostrap server to localhost.  Is there an obvious way to get a KafkaAdmin that has been configured via the properties file (bootstrap and security settings)?

**Comment by massivespace** [comment]
For the benefit of others, I am using spring cloud kafka binder (which appears to be the recommended approach by spring boot), not spring-kafka directly.  The documentation for the kafka binders is out of date in several locations, so maybe I found the wrong configurations, but after multiple tries, I ended up with working properties for brokers at: spring.cloud.stream.kafka.binder.brokers.  

Moving this to spring.kafka.bootstrap-servers allows me to utilize the configuration properties mentioned above.
